Mathlib/RingTheory/Congruence.lean

```diff
@@ -34,17 +34,13 @@ Most of the time you likely want to use the `Ideal.Quotient` API that is built o
 -/
 
 
-/- Note: we can't extend both `AddCon R` and `MulCon R` in Lean 3 due to interactions between old-
-and new-style structures. We can revisit this in Lean 4. (After and not during the port!) -/
 /-- A congruence relation on a type with an addition and multiplication is an equivalence relation
 which preserves both. -/
-structure RingCon (R : Type*) [Add R] [Mul R] extends Setoid R where
-  /-- Ring congruence relations are closed under addition -/
-  add' : ∀ {w x y z}, r w x → r y z → r (w + y) (x + z)
-  /-- Ring congruence relations are closed under multiplication -/
-  mul' : ∀ {w x y z}, r w x → r y z → r (w * y) (x * z)
+structure RingCon (R : Type*) [Add R] [Mul R] extends Con R, AddCon R where
 #align ring_con RingCon
 
+attribute [inherit_doc RingCon] RingCon.toCon RingCon.toAddCon
```

Comment:
Won't this give completely the wrong documentation?

Reply:
I got confused as to why I needed to document this in the first place and just did that. What is a better solution?

Reply:
I think there's an `add_decl_doc` command or similar, with which you can attach the docstrings which were already in the file.